include/gsm0710muxer/muxer_impl.h

```diff
@@ -1420,7 +1424,7 @@ inline const char* Muxer<StreamT, MutexT>::Channel::stateName(ChannelState state
 
 template<typename StreamT, typename MutexT>
 inline void Muxer<StreamT, MutexT>::Channel::transition(ChannelState nState) {
-    GSM0710_LOG_DEBUG(TRACE, "Channel %u transitioning from %s to %s",
+    LOG_DEBUG_C(TRACE, "mux", "Channel %u transitioning from %s to %s",
```

Comment:
@technobly This should work as-is, right? Is the problem that `getLogCategory()` is not available?

Reply:
Here's what it fixes, when you enable muxer debugging with `#define GSM0710_ENABLE_DEBUG_LOGGING (1)` at the top of `muxer_impl.h`
```
✶ brett ✶ ~/code/device-os/main $ make clean all -s PLATFORM=boron APP=tinker MODULAR=n DEBUG_BUILD=y
In file included from network/ncp_client/sara/sara_ncp_client.cpp:20:
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h: In instantiation of 'void gsm0710::Muxer<StreamT, MutexT>::Channel::transition(gsm0710::Muxer<StreamT, MutexT>::ChannelState) [with StreamT = particle::EventGroupBasedStream; MutexT = StaticRecursiveMutex]':
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h:1358:24:   required from 'int gsm0710::Muxer<StreamT, MutexT>::channelTransition(gsm0710::Muxer<StreamT, MutexT>::Channel*, gsm0710::Muxer<StreamT, MutexT>::ChannelState) [with StreamT = particle::EventGroupBasedStream; MutexT = StaticRecursiveMutex]'
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h:296:13:   required from 'int gsm0710::Muxer<StreamT, MutexT>::openChannel(uint8_t, gsm0710::Muxer<StreamT, MutexT>::ChannelDataHandler, void*) [with StreamT = particle::EventGroupBasedStream; MutexT = StaticRecursiveMutex; uint8_t = unsigned char; gsm0710::Muxer<StreamT, MutexT>::ChannelDataHandler = int (*)(const unsigned char*, unsigned int, void*); size_t = unsigned int]'
network/ncp_client/sara/sara_ncp_client.cpp:1551:101:   required from here
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h:32:80: error: cannot call member function 'const char* gsm0710::Muxer<StreamT, MutexT>::getLogCategory() const [with StreamT = particle::EventGroupBasedStream; MutexT = StaticRecursiveMutex]' without object
   32 | #define GSM0710_LOG_DEBUG(_level, _fmt, ...) LOG_DEBUG_C(_level, getLogCategory(), _fmt, ##__VA_ARGS__)
      |                                                                  ~~~~~~~~~~~~~~^~
../services/inc/logging.h:340:49: note: in definition of macro 'LOG_C'
  340 |                 log_message(LOG_LEVEL_##_level, _category, &_attr, NULL, _fmt, ##__VA_ARGS__); \
      |                                                 ^~~~~~~~~
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h:32:46: note: in expansion of macro 'LOG_DEBUG_C'
   32 | #define GSM0710_LOG_DEBUG(_level, _fmt, ...) LOG_DEBUG_C(_level, getLogCategory(), _fmt, ##__VA_ARGS__)
      |                                              ^~~~~~~~~~~
../gsm0710muxer/gsm0710muxer/include/gsm0710muxer/muxer_impl.h:1425:5: note: in expansion of macro 'GSM0710_LOG_DEBUG'
 1425 |     GSM0710_LOG_DEBUG(TRACE, "Channel %u transitioning from %s to %s",
      |     ^~~~~~~~~~~~~~~~~
make[1]: *** [../build/target/hal/platform-13/././network/ncp_client/sara/sara_ncp_client.o] Error 1
make: *** [hal] Error 2
```

Reply:
Ah, I see